webhook_to_fedora_messaging_messages/forgejo/utils.py

```diff
@@ -0,0 +1,146 @@
+# SPDX-FileCopyrightText: 2024-2025 Contributors to the Fedora Project
+#
+# SPDX-License-Identifier: LGPL-3.0-or-later
+
+import json
+
+
+def serialize_body(data: dict) -> str:
+    """
+    Obtain deterministic string conversion of an object
+    """
+    return json.dumps(data, sort_keys=True, separators=(",", ":"))
+
+
+def summarize_repository_event(event_type: str, data: dict) -> str:
+    """
+    Obtain text specification based on event type
+    """
+    if event_type == "push":
+        return _summarize_push_event(data)
+    elif event_type == "fork":
+        return _summarize_fork_event(data)
+    elif event_type == "pull_request":
+        return _summarize_pull_request_event(data)
+    elif event_type == "issues":
+        return _summarize_issues_event(data)
+    elif event_type == "issue_comment":
+        return _summarize_issue_comment_event(data)
+    else:
+        return "Event type not supported"
+
+
+def _summarize_push_event(data: dict) -> str:
+    """
+    Obtain text specification for push event
+    """
+    repository_name = data["repository"]["full_name"]
+    repository_url = data["repository"]["html_url"]
+    pusher = data["pusher"]["full_name"] or data["pusher"]["login"]
+    pusher_url = data["pusher"]["html_url"]
+    branch = data["ref"].split("/")[-1]
+    commits = "\n".join(
+        [
+            f"Commit: {commit['message']} by {commit['author']['name']} ({commit['url']})"
```

Comment:
Possible improvement as mentioned above:
```suggestion
            f"- {commit['id'][:7]}: \"{commit['message'].split('\n')[0]}\" by {commit['author']['name']} ({commit['url']})"
```

Reply:
Nearly there.

One small improvement to that - I think it would be preferrable if we use the following

```suggestion
            f"{commit['id'][0:7]}: {commit['message'].splitlines()[0]} by {commit['author']['name']} ({commit['url']})"  # noqa: E501
```

Tox was complaining when I was using the `'\n'` notation for the `Python 3.11` environment

```
.tox/py311/lib/python3.11/site-packages/_pytest/python.py:493: in importtestmodule
    mod = import_path(
.tox/py311/lib/python3.11/site-packages/_pytest/pathlib.py:587: in import_path
    importlib.import_module(module_name)
/usr/lib64/python3.11/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1204: in _gcd_import
    ???
<frozen importlib._bootstrap>:1176: in _find_and_load
    ???
<frozen importlib._bootstrap>:1147: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:690: in _load_unlocked
    ???                                                                                                                                                                                                                                                                                                                                                                   .tox/py311/lib/python3.11/site-packages/_pytest/assertion/rewrite.py:185: in exec_module
    exec(co, module.__dict__)                                                                                                                                                                                                                                                                                                                                             tests/test_forgejo/test_forgejo.py:7: in <module>
    from webhook_to_fedora_messaging_messages import ForgejoMessageV1
webhook_to_fedora_messaging_messages/__init__.py:5: in <module>
    from .forgejo import ForgejoMessageV1                                                                                                                                                                                                                                                                                                                                 webhook_to_fedora_messaging_messages/forgejo/__init__.py:5: in <module>                                                                                                                                                                                                                                                                                                       from .v1 import ForgejoMessageV1  # noqa: F401
webhook_to_fedora_messaging_messages/forgejo/v1.py:9: in <module>
    from .utils import serialize_body, summarize_repository_event
E     File "/home/fedohide-origin/projects/webhook-to-fedora-messaging-messages/webhook_to_fedora_messaging_messages/forgejo/utils.py", line 45
E       for commit in data["commits"]
E       ^^^
E   SyntaxError: f-string expression part cannot include a backslash
```

I strongly suspect that it would be just as broken for the older versions too but I could not be bothered to test them out.

Reply:
Yeah, `splitlines()` is better indeed, thanks!